### ISSUE TITLE
Export all files from a single source

### ISSUE THREAD

**Issue by sssoleileraaa** [freedomofpress/securedrop-client#550]
## Description

Add an "Export All" option to the source menu so we can give journalists a way to export all files from a single source.

Currently we are using [this format](https://github.com/freedomofpress/securedrop-export#export-folder-structure) for document exports.

Note: https://github.com/freedomofpress/securedrop-client/pull/547 added support for building archives that can contain multiple files

**Comment by eloquence** [comment]
We've considered different strategies for this, such as a the briefcase workflow showcased here:

https://projects.invisionapp.com/share/E7QW28V9HCU#/screens/350508109_Export_Begin_Closed

IMO the complexity of any workflow that allows individual files across sources to be staged for an export (and therefore also be removed from the staging area, and for the staging area to be inspected by the user) is of prohibitive complexity for the beta.

However, I think we should target implementation of an "Export all files/messages" action that can apply at the level of single source, and the user experience of which would be very similar to the single-file export.

**Comment by creviera** [comment]
Allowing queued exports is being tracked in a [separate issue](https://github.com/freedomofpress/securedrop-client/issues/551) than this, which focuses on a single multifile export.

**Comment by eloquence** [comment]
> I think we should target implementation of an "Export all files/messages" action that can apply at the level of single source, and the user experience of which would be very similar to the single-file export.

Currently there is no way to export messages from the client. Copy/paste is disabled, and the messages live in the database, making them difficult to work with even for technical users. For this reason, an "Export all" feature is IMO important to consider as a near-term improvement, to ensure there is at least one supported method to get messages out.

**Comment by ninavizz** [comment]
Most straightforward UX I feel, would be to add bulk export things for each source in the source's "more" menu. I still love the idea of "Briefcase," but to Erik's point above—that's a far more complex thingydoo to execute well, and would require at least a few rounds of user testing to get right and/or validate with users.

![image](https://user-images.githubusercontent.com/8262612/78173610-d3333700-740c-11ea-8a2b-f9f2761c7d40.png)

**Comment by eloquence** [comment]
Yes, that makes sense to me as well. @ninavizz I think we need to start with a more minimal iteration that just offers *Export all (messages + files)* and *Delete all* behavior.

**Comment by creviera** [comment]
> However, I think we should target implementation of an "Export all files/messages" action that can apply at the level of single source, and the user experience of which would be very similar to the single-file export.

That makes sense. I'll de-scope this issue so we can focus on this as an mvp.

**Comment by creviera** [comment]
@gonzalo-bulnes and I have been discussing this issue quite a bit, and I realize that we haven't been adding notes anywhere yet. We agreed that starting with a way to export all files from a source (skipping messages for now) is the first step (prioritizing getting it work end-to-end with the expected folder structure on `sd-devices`, see https://github.com/freedomofpress/securedrop-export#export-folder-structure). Next pieces to get working are:
- Updating the export GUI so that it says "Exporting files from <journalist designation>" instead of "Exporting <filename>"
- Updating the kebab menu to add an "Export all" option
- Integrating the "Download all" option

Once this first iteration is done:
- Export conversation in a single file
- Use journalist designation names for directory names, e.g.
```
├── cytotoxic-payer
│   ├── cytotoxic-payer-conversation.txt
│   ├── interesting_file.doc
│   └── interesting_file\(1\).doc
└── grandiloquent-pasteboard
      └── questionable_file.pdf
```

**Comment by eloquence** [comment]
This was implemented quite a while ago (#1627). :)
